src/core/unit_tests/reaction_ensemble_classes_test.cpp

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright (C) 2021 The ESPResSo project
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* Unit tests for the ReactionEnsemble classes. */
+
+#define BOOST_TEST_NO_MAIN
+#define BOOST_TEST_MODULE ReactionEnsemble classes test
+#define BOOST_TEST_ALTERNATIVE_INIT_API
+#define BOOST_TEST_DYN_LINK
+#include <boost/test/unit_test.hpp>
+
+#include "reaction_ensemble.hpp"
+
+#include "communication.hpp"
+#include "particle_data.hpp"
+
+#include <utils/math/sqr.hpp>
+
+#include <boost/mpi.hpp>
+
+#include <cmath>
+#include <limits>
+#include <memory>
+#include <stdexcept>
+#include <vector>
+
```

Comment:
all the tests need documentation

Reply:
I'm working on it

Reply:
The unit test comments now show the equation of the chemical reactions and explain the different Monte Carlo schemes being tested. The particle types are also now stored in variables instead of hardcoded.